### PR TITLE
For #17847: adding back editCustomSearchEngineTest UI test

### PR DESCRIPTION
**Pull request by sv-ohorvath**
For #17847: Ran 500 times 😅 with this search engine and all were successful. Let's add it back and monitor if it will have the same problem as before.

### CODE REVIEW

**Review by isabelrios — APPROVED** [review]
LGTM

**Comment by sv-ohorvath** [comment]
Test failing now, will have to fix that and I will aks for a new review.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSearchRobot.kt [hunk 1]
```diff
@@ -53,16 +58,47 @@ class SettingsSubMenuSearchRobot {
...
+    }
+
+    fun clickEdit() = onView(withText("Edit")).click()
+
+    fun saveEditSearchEngine() {
+        onView(withId(R.id.save_button)).click()
+        assertTrue(
+            mDevice.findObject(
+                UiSelector().textContains("Saved")
+            ).waitForExists(waitingTime)
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSearchRobot.kt:98]: Here as well

**Comment by sv-ohorvath** [comment]
@isabelrios Please have another look, I made some small changes - see the comments I left.

**Review by isabelrios — APPROVED** [review]
Thanks for fixing this again

**Pull request merged**